### ISSUE TITLE
Gradle plugin doesn't rewrite androidx.* classes after upgrading to AGP 4.2.0

### ISSUE THREAD

**Issue by jschear** [msintuneappsdk/ms-intune-app-sdk-android#69]
**Describe the bug:**
After upgrading our build to use version 4.2.0 of the Android Gradle Plugin, many fewer classes are rewritten by the "mamify" transformation. Before the upgrade, 385 classes were rewritten; after, only 65 are.

In the logs for the `transformClassesWithMamifyTransformFor<variant>` task, I noticed this warning:
```
Configuration specified to include the library androidx.* but it was not seen. Is there a typo in your includeExternalLibraries configuration?
```
... which I believe is pointing to a bug in the plugin, because adding `androidx.*` to `includeExternalLibraries` is exactly what is recommended by the [documentation](https://docs.microsoft.com/en-us/mem/intune/developer/app-sdk-android#androidx-libraries).

**To Reproduce**
1. Update your dependency on the Android Gradle Plugin to 4.2.0 or above.
2. Run `./gradlew my-app:transformClassesWithMamifyTransformForMyBuildVariant`

**Expected behavior:**

The same number of classes should be rewritten; adding `androidx.*` to the Gradle plugin's should transform classes contained in `androidx` libraries. 

**Screenshots and logs:**
```
> Task :app:transformClassesWithMamifyTransformForIntuneDebug

# There are many more statements like this.
Excluding <redacted> because superclass androidx.core.app.ComponentActivity will not be processed
Writing report.
Writing outputs
Configuration specified to include the library androidx.* but it was not seen. Is there a typo in your includeExternalLibraries configuration?
```

**Smartphone (please complete the following information):**
 - N/A (this is a build issue).

**Intune App SDK for Android (please complete the following information):**
- What version of the Intune SDK are you using? Are you using the latest version? 
  - 7.2.2; yes.
- What platform is your app based in (Java, Xamarin based, Cordova, etc)? 
  - Java/Kotlin.
- For errors during build, does the app build without Intune SDK integration?
  - Yes.
- For errors post build, does the app launch without being Intune SDK integrated?
  - Yes.
- Who is the customer?
- Do you see a trend with it only being reproduced on a specific device?
  - No.

**Comment by snijsure** [comment]
I switched to gradle-7.0.2 from 6.5.1 and Android Gradle plugin 4.1.3 to 4.2.0. m"Espresso" test shown below, fails, when using the latest version of the MAM library.

    ```
@Test
    fun testIsMamActivity() {
        var isSubclassOfMamActivity = false
        // ACT
        val activity = activityTestRule.launchActivity(getHomeIntent())

        // Walk-up the class hierarchy and make sure we find MAMActivity
        var superClass: Class<*>? = activity.javaClass
        while (superClass != null) {
            if (superClass.name == MAMActivity::class.java.name) {
                isSubclassOfMamActivity = true
                break
            }
            superClass = superClass.superclass
        }

        // ASSERT
        assertThat(isSubclassOfMamActivity).isTrue
    }
```

**Comment by snijsure** [comment]
Hmm, this could be because gradle 7.0 requires us to use android gradle plugin 7.0.0-beta03 (or whichever is the latest?)

**Comment by joshbohls** [comment]
We are seeing the same issue. Has anyone at @msintuneappsdk or in the dev community found a solution?

**Comment by jschear** [comment]
@joshbohls this has been resolved in the latest release, 7.6.0 ([changelog](https://github.com/msintuneappsdk/ms-intune-app-sdk-android/releases/tag/7.6.0)):
> The build plugin now automatically includes all external libraries when used with Android Gradle Plugin 4.2 and higher. These versions no longer expose the library names to the Transform API which our plugin is built on. The includeExternalLibraries configuration option will be removed in future MAM SDK 8.0.

Though in my testing, specifying `androidx.*` for `includeExternalLibraries` was still necessary to get the transform to run on classes brought in by androidx libraries, so I think the release notes aren't quite correct.

**Comment by meghandaly** [comment]
Resolved in the latest release, [version 7.6.0](https://github.com/msintuneappsdk/ms-intune-app-sdk-android/releases/tag/7.6.0).
